src/blob/core.rs

```diff
@@ -557,7 +563,7 @@ impl RawRecords {
         }
     }
 
-    async fn read_current_record_header(&mut self) -> Result<RecordHeader> {
+    async fn read_current_record(&mut self, read_data: bool) -> Result<(RecordHeader, Vec<u8>)> {
```

Comment:
It is better to return `Result<(RecordHeader, Option<Vec<u8>>)>` to explicitly indicate that data was not read